src/util/debug.c

```diff
@@ -287,7 +290,31 @@ void sss_vdebug_fn(const char *file,
          * searchable.
          */
         va_copy(ap_fallback, ap);
-        ret = journal_send(file, line, function, level, format, ap);
+        if (debug_chain_id > 0) {
+            ret = asprintf(&chain_id_format, "[RID#%lu] ", debug_chain_id);
```

Comment:
mem allocations are quite expensive.
You can avoid this using fixed size buffer (maybe even static since all sssd processes are single-thread)
Or even better just add a constant to `new_size` and print `"[RID#%lu] %s" in a single `snprintf()`...